### PR TITLE
Add Mesos Installer

### PR DESCRIPTION
**Pull request by mattboehm**
Running `ansible-playbook provision-mesos.yml` will install onto mesos

* Read many vars from the env
* Don't use sudo
* Only run ckan/ace tasks
* Run apache in the foreground
* Store all files in MESOS_SANDBOX (and store ckan user uploads on the network drive)

### CODE REVIEW

**Comment by mattboehm** [comment]
Everything should be working now but before merging, I'd like to do a complete vagrant destroy/reinstall for both the traditional and mesos installations to ensure that nothing broke. If anyone else would like to try the mesos installer, I've added instructions to the readme.

File: README.md
Comment (tanderegg) [inline README.md]: In the Marathon UI, if you go to the app and "instances", it will list the running instances and the IP address they are deployed to, which is probably the easiest way to see which Agent it is running on.

File: README.md
Comment (tanderegg) [inline README.md]: Note that you can also view the sandbox through the Mesos UI, if you go to 10.0.1.31:5050 for example, click on the "sandbox" link next to the running task, and you can browse it in the browser.

File: marathon_app.json
Comment (tanderegg) [inline marathon_app.json]: Normally I try to use RAM as a base 2 value, i.e. 128 instead of 100, but this is not strictly necessary.  Also, we don't really use disk isolation yet, so setting "disk" to 0 is fine for now.

File: marathon_app.json
Comment (tanderegg) [inline marathon_app.json]: It might be cleaner just to put ANSIBLE_CONFIG in the "env" block below rather in the command itself.

File: marathon_app.json
Comment (tanderegg) [inline marathon_app.json]: Having the port here explicitly shouldn't be needed, the "portIndex" 0 below will explicitly use the port assigned to this instance of the app.  Is Apache setup to use $PORT0?

File: marathon_app.json
Comment (tanderegg) [inline marathon_app.json]: The "fetch" block isn't strictly needed, as it is autogenerated by Marathon from the "uris" block.  I usuall remove it from the JSON definition, as things like "cache" are dynamically set depending on how Marathon/Mesos are configured.

**Comment by mattboehm** [comment]
I've tested both forms of installations and am having Valeri run through the instructions now to try it for himself.

File: provisioners/group_vars/mesos
Comment (tanderegg) [inline provisioners/group_vars/mesos]: This should probably be {{ lookup('env', 'PORT0') }}, to pull in the port from Marathon, no?

**Review by tanderegg — CHANGES_REQUESTED** [review]

File: marathon_app.json
Comment (tanderegg) [inline marathon_app.json]: @mattboehm This should be changed to the primary repo before merging

Comment (tanderegg) [reply marathon_app.json]: Ah, and actually this should be the "uris" key not the "fetch" key.  Basically, "fetch" is autogenerated by Marathon for each item in the "uris" lists.  So you just need:
```
"uris": {
  "https://github.com..."
}
```

File: provisioners/roles/ckan/tasks/main.yml
Comment (tanderegg) [inline provisioners/roles/ckan/tasks/main.yml:82]: @mattboehm You could also just do `{{ckan_venv}}/bin/python -m ...` instead of activating the virtualenv

File: provisioners/roles/ckan/tasks/main.yml
Comment (tanderegg) [inline provisioners/roles/ckan/tasks/main.yml:85]: Same here, you could explicitly use the virtualenv's python, and also potentially use Ansible's "environment" option like here: https://github.com/cfpb/aurora/blob/2ad6bd89dec788cc787f9b4883e82d0bf25ff25c/deploy/roles/sqitch/tasks/main.yml#L53, but with ansible_env.PATH to retrieve the target's PATH variable.

File: provisioners/roles/ckan/tasks/main.yml
Comment (tanderegg) [inline provisioners/roles/ckan/tasks/main.yml]: Should we remove the commented out lines at this point?

**Review by tanderegg — CHANGES_REQUESTED** [review]

**Comment by tanderegg** [comment]
@mattboehm Just some minor stuff, thanks!

File: provisioners/roles/ckan/tasks/main.yml
Comment (mattboehm) [reply provisioners/roles/ckan/tasks/main.yml:82]: I tend to prefer this since it updates a bunch of environment variables and I'm never quite sure if a pip install is going to shell out to some command that depends on those variables being set.

File: provisioners/roles/ckan/tasks/main.yml
Comment (mattboehm) [reply provisioners/roles/ckan/tasks/main.yml:85]: ansible_env.PATH no longer works without further fiddling: https://github.com/ansible/ansible/issues/14655

The workarounds in that thread seem flaky and complicated, so for now I'm going to revert to setting the path inline but am open to suggestions.

**Review by tanderegg — APPROVED** [review]

**Comment by tanderegg** [comment]
@mattboehm Is the [WIP] label still applicable, or is this good to go from your end?  Looks correct to me, so I'm ready to merge if you are.

**Comment by tanderegg** [comment]
@mattboehm bump

**Comment by mattboehm** [comment]
@tanderegg whoops, I've removed the WIP label. We've been using this in prod for a while now, so it should be good to merge.

**Comment by tanderegg** [comment]
@mattboehm Cool, that is what I thought, just wanted to be prudent :)

**Pull request merged**
